package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosFailoverAwareRRPolicy.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) Microsoft. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.azure.cosmos.cassandra;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import com.datastax.driver.core.HostDistance;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.policies.LoadBalancingPolicy;
+import com.google.common.collect.AbstractIterator;
+import javafx.util.Pair;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * CosmosDB failover aware Round-robin load balancing policy.
+ * This policy allows the user to seamlessly failover the default write region.
+ * This is very similar to DCAwareRoundRobinPolicy, with a difference that
+ * it considers the nodes in the default write region at distance LOCAL.
+ * All other nodes are considered at distance REMOTE.
+ * The nodes in the default write region are retrieved from the global endpoint
+ * of the account, based on the dns refresh interval of 60 seconds by default.
+ */
+public class CosmosFailoverAwareRRPolicy implements LoadBalancingPolicy
```

Comment:
Coding guidelines: Please reformat this file to use K&R brace/indentation style, not the BSD style commonly used with C#.